Edgar-Alandete/challenges/game-of-life/src/gameOfLife.js

```diff
@@ -0,0 +1,181 @@
+const BLINKER_HEIGHT = 5;
+const BLINKER_WIDTH = 5;
+
+const TOAD_WIDTH = 6;
+const TOAD_HEIGHT = 6;
+
+const DEAD = 0;
+const ALIVE = 1;
+
+let newBoard = [];
+let intervalId = null;
+
+const updateCellStatus = (isAlive) => {
+  return isAlive ? ALIVE : DEAD;
+};
+
+const applyRules = (cell, neighbours, isAlive) => {
+  let cellStatus = DEAD;
+  if (isAlive(cell)) {
+    cellStatus = updateCellStatus(neighbours === 2 || neighbours === 3);
+  } else {
+    cellStatus = updateCellStatus(neighbours === 3);
+  }
+
+  return cellStatus;
+};
+
+const cellExists = (cell) => {
+  return cell !== undefined;
+};
+
+const hasNeighbour = (row, column) => {
+  return cellExists(row) && cellExists(column);
+};
+
+const getNeighbours = (board, row, column) => {
+  let neighbours = 0;
+  if (
+    hasNeighbour(board[row - 1], board[column - 1]) &&
+    isAlive(board[row - 1][column - 1])
+  ) {
+    neighbours++;
+  }
+  if (
+    hasNeighbour(board[row - 1], board[column]) &&
+    isAlive(board[row - 1][column])
+  ) {
+    neighbours++;
+  }
+  if (
+    hasNeighbour(board[row], board[column - 1]) &&
+    isAlive(board[row][column - 1])
+  ) {
+    neighbours++;
+  }
+  if (
+    hasNeighbour(board[row - 1], board[column + 1]) &&
+    isAlive(board[row - 1][column + 1])
+  ) {
+    neighbours++;
+  }
+  if (
+    hasNeighbour(board[row], board[column + 1]) &&
+    isAlive(board[row][column + 1])
+  ) {
+    neighbours++;
+  }
+  if (
+    hasNeighbour(board[row + 1], board[column - 1]) &&
+    isAlive(board[row + 1][column - 1])
+  ) {
+    neighbours++;
+  }
+  if (
+    hasNeighbour(board[row + 1], board[column]) &&
+    isAlive(board[row + 1][column])
+  ) {
+    neighbours++;
+  }
+  if (
+    hasNeighbour(board[row + 1], board[column + 1]) &&
+    isAlive(board[row + 1][column + 1])
+  ) {
+    neighbours++;
+  }
+
+  return neighbours;
+};
+
+const updateFigure = (board, isAlive) => {
+  let updatedBoard = [];
+  board.forEach((row) => {
+    console.log("row" + row);
+    //updatedBoard = [...row];
+  });
+
+  board.forEach((row, rowIndex) => {
+    const newRow = [];
+    row.forEach((cell, cellIndex) => {
+      const neighbours = getNeighbours(board, rowIndex, cellIndex);
+      const cellStatus = applyRules(cell, neighbours, isAlive);
+      newRow.push(cellStatus);
+    });
+    updatedBoard.push(newRow);
+  });
+
+  console.log(updatedBoard);
+  newBoard = updatedBoard;
+
+  return updateTableDom(updatedBoard);
+};
+const isAlive = (cell) => cell === ALIVE;
+
+const updateTableDom = (figure) => {
+  let tableRows = document.getElementsByTagName("tr");
+
```

Comment:
cuidado con estos espacios

Reply:
creo que este espacio esta bien porque esta separando al declaración de variables del for

Reply:
Para mi los espacios tienen que ir definiendo las diferentes partes de una funcion por bloques: 

Variables
Bloques de codigo. 
Por cada bloque de codigo un intro para que se vea la diferencia